### PR TITLE
DMBM 705 Legal power

### PR DESCRIPTION
**Pull request by madz-lw**
- This PR implements the legal power stage of the form which contains radio buttons and text areas for user input.
- The 'yes' radio button triggers an additional user input area, whereas the negative buttons do not.
- There is a LegalPowerStep type that holds the two values that form part of the legal power value of the form data. First, a decision which represents the yes/no/we don't know radio buttons, and an explanation text area, which is only available to the 'yes' response.
- The template for the Get Legal Advice window is also in this PR but the routing is yet to be set up whilst the logic is being decided.

### CODE REVIEW

File: src/views/acquirer/legal-power.njk [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            },
+            hint: {
+                text: "For example, through common law or royal prerogative."
+            },
+            items: 
+            [
+            {
+                value: "yes",
+                text: "Yes, we believe we have the legal power to do this",
+                checked: savedValue['legal-power'].decision === 'yes',
```
Comment (kjoshi) [inline src/views/acquirer/legal-power.njk:38]: The `savedValue` that gets passed into the template looks like this:
`value: { decision: 'yes', explanation: 'something' }`
So I think this needs to be changed to:
`checked: savedValue.decision === 'yes',`

File: src/views/acquirer/legal-power.njk [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+                conditional: {
+                    html: legalPowerDecision
+                }
+            },
+            {
+                value: "no",
+                text: "No, we don't have the legal power to request this",
+                checked: savedValue['legal-power'].decision === "no"
```
Comment (kjoshi) [inline src/views/acquirer/legal-power.njk:47]: Same as above, I think you need to remove the `['legal-power']` bit.

File: src/views/acquirer/legal-power.njk [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            },
+            {
+                value: "we don't know",
+                text: "We don't know",
+                checked: savedValue['legal-power'].decision === "we don't know"
```
Comment (kjoshi) [inline src/views/acquirer/legal-power.njk:52]: Same as above.

File: src/views/acquirer/legal-power.njk [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      <span class="govuk-caption-l">Request ID: {{requestId}}</span>
+      <form method="POST" id="legalPowerForm">
+        {% set legalPowerDecision %}
+            {{ govukTextarea({
+                id: "legal-power-textarea",
+                name: "legal-power-textarea",
+                label: {
+                    text: "What legal power will you use?"
+                },
+                value: savedValue['legal-power'].explanation
```
Comment (kjoshi) [inline src/views/acquirer/legal-power.njk:18]: Same comment as below, I think you need to remove the `['legal-power']` part

**Review by kjoshi — CHANGES_REQUESTED** [review]
Thanks Maddie, just found one small thing to update so that the saved values are displayed properly in the form.

File: src/views/acquirer/legal-power.njk (hunk 1 above)
Comment (madz-lw) [reply src/views/acquirer/legal-power.njk:38]: Good spot! Thank you, I'll push the changes in a commit shortly 😊

**Review by kjoshi — APPROVED** [review]
Looks good, thanks 👍

**Pull request merged**